source/dub/internal/utils.d

```diff
@@ -325,6 +325,92 @@ ubyte[] download(URL url, uint timeout = 8)
 	return download(url.toString(), timeout);
 }
 
+/**
+	Downloads a file from the specified URL with retry logic.
+
+	Downloads a file from the specified URL with up to 3 tries on failure
+	Throws: `Exception` if the download failed or `HTTPStatusException` on "unrecoverable failures" such as 404 not found.
+	Otherwise might throw anything else that `download` throws.
+	See_Also: download
+**/
+void retryDownload(URL url, NativePath filename)
+{
+	foreach(i; 0..3) {
```

Comment:
Probably a good idea to make n configurable if we already generalize this function?

Reply:
done